tests/unit/services/apigateway/test_handler_integration_request.py

```diff
@@ -0,0 +1,265 @@
+
+        # If a matching template is found, it should use it
+        context.resource_method["methodIntegration"]["requestTemplates"] = {
+            "application/json": '{"foo":"bar"}'
+        }
+        integration_request_handler(context)
+        assert context.integration_request["body"] == b'{"foo":"bar"}'
+
+        # If no template were created, it should passthrough
+        context.resource_method["methodIntegration"]["requestTemplates"] = {}
+        integration_request_handler(context)
+        assert context.integration_request["body"] == b""
+
+    def test_default_template(
+        self, integration_request_handler, default_invocation_request, create_context
+    ):
+        request = default_invocation_request
+        context = create_context(request=request)
+
+        # if no matching template, use the default
+        context.resource_method["methodIntegration"]["requestTemplates"] = {
+            "$default": '{"foo":"bar"}'
+        }
+        integration_request_handler(context)
+        assert context.integration_request["body"] == b'{"foo":"bar"}'
+
+        # If there is a matching template, use it instead
+        context.resource_method["methodIntegration"]["requestTemplates"] = {
+            "$default": '{"foo":"bar"}',
+            "application/json": "Matching Template",
+        }
+        integration_request_handler(context)
+        assert context.integration_request["body"] == b"Matching Template"
+
+    def test_request_parameters(
+        self, integration_request_handler, default_invocation_request, create_context
+    ):
+        context = create_context(request=default_invocation_request)
+        context.resource_method["methodIntegration"]["requestParameters"] = {
+            "integration.request.path.path": "method.request.path.path",
+            "integration.request.querystring.qs": "method.request.querystring.qs",
+            "integration.request.header.header": "method.request.header.header",
+        }
+        context.resource_method["methodIntegration"]["uri"] = "https://example.com/{path}"
+        integration_request_handler(context)
+        # TODO this test will fail when we implement uri mapping
+        assert context.integration_request["uri"] == "https://example.com/{path}"
+        assert context.integration_request["query_string_parameters"] == {"qs": "test-qs-value"}
+        assert context.integration_request["headers"] == {"header": "test-header-value"}
+
+    def test_request_override(
+        self, integration_request_handler, default_invocation_request, create_context
+    ):
+        context = create_context(request=default_invocation_request)
+        context.resource_method["methodIntegration"]["requestParameters"] = {
+            "integration.request.path.path": "method.request.path.path",
+            "integration.request.querystring.qs": "method.request.querystring.qs",
+            "integration.request.header.header": "method.request.header.header",
+        }
```

Comment:
what happens if you were to use the following:
```python
"integration.request.querystring.qs": "method.request.multivaluequerystring.qs",
```
and then overrides? does it just erase the list and sets only one value after the override? I'd say so, that would be pretty logic!

Reply:
Good question.. I should play with that a bit!

Reply:
I have looked and yes the override always overrides. I was not able to create a `join` scenario. I have also added a test for it